cores/esp32/esp32-hal-periman.c

```diff
@@ -8,9 +8,12 @@
 #include "esp32-hal-periman.h"
 #include "esp_bit_defs.h"
 
-typedef struct {
+typedef struct ATTR_PACKED {
 	peripheral_bus_type_t type;
+	const char* extra_type;
 	void * bus;
+	int8_t bus_num;
+	int8_t bus_channel;
```

Comment:
Shouldn't `extra_type` be changed to `description`? I couldn't see other usage for it...

Reply:
it is an extra type. That is what is being used for. We added it for the cases where GPIO is being used for some specific task (like CS, RST, DC, etc.) and cases where different peripherals are used for some other specific task/driver. Hypothetic example would be implementing NeoPixel over RMT/Else and setting the extra type as `NEOPIXEL_OUT`